src/Content/Monaco.Template.Common.Application/DTOs/Extensions/BaseTypeExtensions.cs

```diff
@@ -0,0 +1,32 @@
+using Monaco.Template.Common.Domain.Model;
+using Monaco.Template.Common.Domain.Model.Contracts;
+using System.Linq.Expressions;
+
+namespace Monaco.Template.Common.Application.DTOs.Extensions;
+
+public static class BaseTypeExtensions
+{
+	public static TResult? Map<TEntity, TResult>(this TEntity? value) 
+		where TEntity : Enumeration, IReferential
```

Comment:
I think you can spare the `IReferential` constraint, to be able to do this for *any* `Enumeration` entity, not only those that are also implementing `IReferential`.

```suggestion
		where TEntity : Enumeration
```